ecs-agent/tmds/handlers/fault/v1/handlers/handlers.go

```diff
@@ -36,56 +51,282 @@ func FaultNetworkFaultPath(fault string) string {
 		utils.ConstructMuxVar(v4.EndpointContainerIDMuxName, utils.AnythingButSlashRegEx), fault)
 }
 
-// TODO
 func (h *FaultHandler) StartNetworkBlackholePort() func(http.ResponseWriter, *http.Request) {
 	return func(w http.ResponseWriter, r *http.Request) {
+		var request types.NetworkBlackholePortRequest
+		requestType := fmt.Sprintf(startFaultRequestType, types.BlackHolePortFaultType)
+
+		// Parse the fault request
+		err := decodeRequest(w, &request, requestType, r)
+		if err != nil {
+			return
+		}
+		// Validate the fault request
+		err = validateRequest(w, request, requestType)
+		if err != nil {
+			return
+		}
+		requestStr, _ := request.ToString()
+		logger.Debug("Successfully parsed fault request payload", logger.Fields{
+			field.Request: requestStr,
+		})
+
+		// Obtain the task metadata via the endpoint container ID
+		// TODO: Will be using the returned task metadata in a future PR
+		_, err = getTaskMetadataForFault(w, h.AgentState, requestType, r)
+		if err != nil {
+			return
+		}
+
+		// TODO: Check status of current fault injection
+		// TODO: Invoke the start fault injection functionality if not running
+
+		responseBody := types.NewNetworkFaultInjectionResponse("running", "")
```

Comment:
nit: There are two kinds of `NetworkFaultInjectionResponse` structs we create - a success response and a failure response. If we introduce separate functions for each of these two cases the code would become more readable. For example, `NewNetworkFaultInjectionSuccessResponse("running")` and `NewNetworkFautlInjectionErrorResponse(err)`.

Reply:
>NewNetworkFaultInjectionSuccessResponse("running")

This is not always the success response.  That's fair, I can make `New` functions for these...